src/OpenSage.Game/Logic/Object/Behaviors/SpawnBehavior.cs

```diff
@@ -44,9 +44,25 @@ private void SpawnUnit()
                 slavedUpdate.Master = _gameObject;
             }
 
+            if (!TryTransformViaProductionExit(spawnedObject) &&
+                !TryTransformViaOpenContainer(spawnedObject))
+            {
+                throw new Exception("Unable to set spawn point for spawned unit");
+            }
+        }
+
+        private bool TryTransformViaProductionExit(GameObject spawnedObject)
+        {
+            _productionExit ??= _gameObject.FindBehavior<IProductionExit>();
+
+            if (_productionExit == null)
+            {
+                return false;
+            }
+
             if (_productionExit != null)
             {
-                spawnedObject.SetTranslation(_gameObject.ToWorldspace(_productionExit.GetUnitCreatePoint()));
```

Comment:
This was most likely done for Bfme reasons. Could be the case that we need a dedicated implementation depending on the game

Reply:
Would you be able to test and see if this pr would break bfme? My suspicion is that the GetUnitCreatePoint method changed at some point to return worldspace automatically since that's probably how it was being used everywhere, but I haven't checked.

Reply:
@charliefoxtwo sadly BFME is quite broken right now, so there would be a lot of fixes required in order to test things again. In my opinion we should focus on Generals right now, because testing many games at the same time is quite time consuming